framework/include/outputs/Checkpoint.h

```diff
@@ -15,6 +15,19 @@
 
 #include <deque>
 
+/**
+ * Shortcut for determining what type of autosave this checkpoint is.
+ * NONE: Not an autosave checkpoint
+ * MODIFIED_EXISTING: We took an existing checkpoint and enabled autosave checks on it.
+ * SYSTEM_AUTOSAVE: These checkpoints run in the background and output only when sent a signal.
+ */
+enum AUTOSAVE_TYPE : unsigned short
```

Comment:
Naming conventions for types say we should use camel case: `AutoSave` or `AutoSaveType`. The enumeration values can remain all caps though.